gender_novels/analysis/gender_pronoun_freq_analysis.py

```diff
@@ -62,6 +62,17 @@ def subject_vs_object_pronoun_freqs(corp):
 
     return result_tuple
 
+def sub_pronouns_gender_comp(corp):
```

Comment:
This function needs a bit of work before we can merge it. 
- name: `subject_pronouns_gender_comparison` is longer but more immediately descriptive
- generality: can you write the function so that it accepts something like a `subject_gender` to calculate? That way, it can be used to calculate relative male and female subject frequencies without changing any of the code.
- documentation: what does it do?
- doc test: how to use it and does it work?

Reply:
I'm confused on what you are asking here. I understand the documentation bit - sorry, that was an oversight. Will do. What do you mean by:

generality: can you write the function so that it accepts something like a subject_gender to calculate? That way, it can be used to calculate relative male and female subject frequencies without changing any of the code.

There's already a function above it called subject_vs_object_pronoun_freqs that calculates the percent of pronouns of one gender that are subject pronouns. This function looks at the total number of subject pronouns and calculates the portion of those that are male and female. This particular function is highly biased to the proportion of total pronouns of each gender, and I realized that when I wrote it. I just wanted to make sure I wasn't overlooking anything. I.e., if this function returned significantly different numbers than the function that gets proportions for total pronouns, there would be something different about subject pronouns than pronouns in general. That was my thinking anyway.

Reply:
Sure. I should have been clearer. The function is very useful. However, it always returns the frequency of female subjects and it would be nice if we could also directly get the frequency of male subjects. It's a mathematically simple operation but there is nothing in the function that would tell me that I'm getting back the frequency of female subjects rather than both. 
So the change could be: 
```
def sub_pronouns_gender_comp(corp, subject_gender):

    relative_frequeny_of_female_subjects = {}
    relative_frequency_of_male_subjects = {}

    for book in corp.novels:
        he = book.get_word_freq('he')
        she = book.get_word_freq('she')
        
        relative_frequency_of_female_subjects[book] = (she) / (he + she)
        relative_frequency_of_male_subjects[book] = (he) / (he + she)
    
    if subject_gender == 'female':
        return relative_frequency_of_female_subjects
    if subject_gender == 'male':
        return relative_frequency_of_male_subjects
```